napari/components/dims.py

```diff
@@ -276,6 +279,24 @@ def not_displayed(self) -> Tuple[int, ...]:
     def displayed_order(self) -> Tuple[int, ...]:
         return tuple(argsort(self.displayed))
 
+    # TODO: I'm not sure if this is the correct way to bind it. I fear this might cause some event miss fireing.
+    @property
+    def n_moveable_dims(self) -> int:
+        """Int: Number of dimensions (from the back) which can be reordered / rolled"""
+        nmdims = get_settings().application.n_moveable_dims
+        # if the requested numer of roallable dimensions exceed the number of dimensions clip to ndim
+        if nmdims > self.ndim:
+            return self.ndim
+        return nmdims
+
+    @n_moveable_dims.setter
```

Comment:
This isn't used anywhere in this PR, which might explain why you're having problems with the event associated with this mutable property. `EventedModel` can support events for derived attributes using `__field_dependents__`, but I think that won't work here since one dependency is on the global settings.